src/fileio.cpp

```diff
@@ -977,55 +973,67 @@ bool DoScanWorkingDirectory()
 	return _searchpaths[SP_PERSONAL_DIR] != tmp;
 }
 
+/**
+ * Gets the home directory of the user.
+ * May return an empty string in the unlikely scenario that the home directory cannot be found.
+ * @return User's home directory
+ */
+static std::string GetHomeDir()
+{
+	std::string homedir;
+#ifdef __HAIKU__
+	BPath path;
+	find_directory(B_USER_SETTINGS_DIRECTORY, &path);
+	homedir = path.Path();
+#else
+	char *home_env = getenv("HOME");
+	if (home_env != nullptr) {
+		homedir = home_env;
+	} else  {
+		const struct passwd *pw = getpwuid(getuid());
+		if (pw != nullptr) {
+			homedir = pw->pw_dir;
+		}
+	}
+#endif
+	return homedir;
+}
+
 /**
  * Determine the base (personal dir and game data dir) paths
  * @param exe the path to the executable
  */
 void DetermineBasePaths(const char *exe)
 {
 	std::string tmp;
-#if defined(WITH_XDG_BASEDIR) && defined(WITH_PERSONAL_DIR)
-	const char *xdg_data_home = xdgDataHome(nullptr);
-	tmp = xdg_data_home;
-	tmp += PATHSEP;
-	tmp += PERSONAL_DIR[0] == '.' ? &PERSONAL_DIR[1] : PERSONAL_DIR;
-	free(xdg_data_home);
+#ifdef USE_XDG
+	std::string homedir = GetHomeDir();
+	const char *xdg_data_home = getenv("XDG_DATA_HOME");
+	if (xdg_data_home != nullptr) {
+		tmp = xdg_data_home;
+		tmp += PATHSEP;
+		tmp += PERSONAL_DIR[0] == '.' ? &PERSONAL_DIR[1] : PERSONAL_DIR;
+	} else if (!homedir.empty()) {
+		tmp = homedir;
+		tmp += PATHSEP ".local" PATHSEP "share" PATHSEP;
+		tmp += PERSONAL_DIR[0] == '.' ? &PERSONAL_DIR[1] : PERSONAL_DIR;
```

Comment:
I always wondered if this is such a good idea or not .. basically, we kinda abuse `PERSONAL_DIR` to get the name of the game to use with XDG ... I really wonder if we shouldn't just use a more generic variable that says "openttd", instead of this magic which makes any sane person go: WHAT THE FUCK, the first time they see it :P Might not be for this PR .. but this feels wrong and nasty :D (but that was already the case ;))

Reply:
I agree, but PERSONAL_DIR is used in OS-agnostic code (~/.openttd & ~/Documents/OpenTTD & <macos>). So, doesn't particularly apply SP_PERSONAL_DIR_XDG, where PERSONAL_DIR will (almost) always begin with a '.', but the SP_PERSONAL_DIR might not